### PR TITLE
Ticket4264 reflectometry ordered positions

### PR DESCRIPTION
**Pull request by SophieKSTFC**
### Description of work

This ticket is part of the work to provide the parameters found on the 'Components: Positions relative to beam' screen on the opi in the same order in which they are found in the config. The work also provides an indication as to whether the parameter is in the mode or not. 

### To test

[#4264](https://github.com/ISISComputingGroup/IBEX/issues/4264)

### Acceptance criteria

- [x] All parameters have an IN_MODE PV.

- [x] When the mode is changed, the appropriate parameters in that mode are set to being IN_MODE = YES.

- [x] The parameters retain the order in which they are provided in the config on the opi. 

---

#### Code Review

- [x] Is the code of an acceptable quality?
- [x] Has the author taken into account the multi-threaded nature of the code?
- [x] Have the changes been documented in the [release notes](https://github.com/ISISComputingGroup/IBEX/wiki/ReleaseNotes_Dev). If so, do they describe the changes appropriately?
- [x] Has the [manual system tests spreadsheet](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Manual-system-tests) been updated?

### Functional Tests

- [ ] Do changes function as described? Add comments below that describe the tests performed.

### Final steps

- [ ] Reviewer has updated the submodule in the main EPICS repo? See **Reviewing work for the subModules of EPICS** in the [Git workflow](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Git-workflow) page for details.
- [x] Reviewer has moved the [release notes](https://github.com/ISISComputingGroup/IBEX/wiki/ReleaseNotes_Dev) entry for this ticket in the "Changes merged into master" section

### CODE REVIEW

File: ReflectometryServer/ChannelAccess/pv_manager.py [hunk 1]
```diff
@@ -246,27 +252,37 @@ def _add_all_parameter_pvs(self, param_types):
...
         """
         try:
             param_alias = create_pv_name(param_name, self.PVDB.keys(), "PARAM", limit=10)
             prepended_alias = "{}:{}".format(PARAM_PREFIX, param_alias)
-            if BeamlineParameterGroup.TRACKING in group_names:
-                self._tracking_positions[prepended_alias] = param_name
+           
+            fields = PARAMS_FIELDS_BEAMLINE_TYPES[param_type]
+            # generate a dictionary to store metadata about tracking positions
+            item = {}
```
Comment (ThomasLohnert) [inline ReflectometryServer/ChannelAccess/pv_manager.py]: `item` is not very descriptive - perhaps `param_info` instead?

Comment (SophieKSTFC) [reply ReflectometryServer/ChannelAccess/pv_manager.py]: renamed in latest commit

File: ReflectometryServer/ChannelAccess/pv_manager.py [hunk 2]
```diff
@@ -185,8 +191,8 @@ def __init__(self, param_types, mode_names, status_codes):
         """
 
         self.PVDB = {}
-        self._params_pv_lookup = {}
-        self._tracking_positions = {}
+        self._params_pv_lookup = OrderedDict()
+        self._tracking_positions = []
```
[the remark this answers is not in the document]
Comment (SophieKSTFC) [reply ReflectometryServer/ChannelAccess/pv_manager.py]: renamed in latest commit, see also https://github.com/ISISComputingGroup/ibex_gui/pull/900

File: ReflectometryServer/reflectometry_server.py [hunk 3]
```diff
@@ -66,6 +66,8 @@
 ioc_data_source.insert_ioc_start("REFL", os.getpid(), sys.argv[0], pv_db.PVDB, REFLECTOMETRY_PREFIX)
 
 logger.info("Reflectometry IOC started")
+
+logger.info(pv_db.PVDB["TRACKING_AXES"])
```
[the remark this answers is not in the document]
Comment (SophieKSTFC) [reply ReflectometryServer/reflectometry_server.py]: removed in latest commit

**Pull request merged**